### PR TITLE
UCP/ADDRESS: Simplify unpack code by not counting addresses

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 20-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/889feb6159771501e94a05c5425d447b24ad714d/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/a457a4d51d17fad611a915c5a6595ca86b634cdb/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/f758e854793b4f8a27f4c6e5c09e746664642b9b/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/00341f10ccccb415424062b83e98810b29da1e0c/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/1bc43fc83316944755e9f670d1b29aafd5649737/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/ccbb3c73cf7a52d0db8fb2652cddfe8d0f3d5d7d/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/b446132cbb5161b4caeda069fdaa1659d64f62c2/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/faa968243bba628ea2fcf52cbc8c2029934f77c2/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/d409c862271d1f4a5bdc740038c80fb11b6df385/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/896e8993030049f063cb55beca0ba2d07698e82f/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/bb48bc414bebc4dbb740d2bcf60b3cef15971e07/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/00f31eebe4a4514482144c754114e09e9474dc5f/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/0d2ef8dff4a455c693f5da579071f533d95fc019/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/4051158917dd1d530a87019d79062a125edade66/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/44c4783974b2e06a2e4ecf952ae446611487f651/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/ab40461c84f247e42b57a1aadddbca5e4698009d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/cab82519a9a7c51037b5df468e19df5e409a47ba/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/b914e414dc80c491bd1d27c0a55fa5b92324537a/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/c2787146bca761dc127fc1d396448ea144a008d1/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/98003f28c79354b84206b3ec9abc0650ad7b1c1b/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/3cd1a0de736214b886365fdcd6d4184931e41b4a/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/e7b5884724f73742533c416592de1139b7508dcc/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/278fe8423529b17a463519131ed52ccfdb9765bd/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/46884a448ec743064bcf6a6f899f6dddf66403cb/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb26c9eba29ed6ebe31aeb7e4d52ecf8/raw/37381caa673db974dd60d97ce3898ac21ef45d11/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@brminich @hoopoepg can you pls review?

File: src/ucp/wireup/address.c [hunk 1]
```diff
@@ -991,11 +950,9 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
         ++dev_index;
     } while (!last_dev);
 
-    ucs_assert((unsigned)(address - address_list) == address_count);
-
-out:
-    unpacked_address->address_count = address_count;
+    unpacked_address->address_count = address - address_list;
     unpacked_address->address_list  = address_list;
+out:
```
Comment (brminich) [inline src/ucp/wireup/address.c]: minor: looks like no need in this label. Now used on line 898 only (can just return there)

**Comment by yosefe** [comment]
> wouldn't it be a noticeable mem consumption increase in case of simultaneous connections on a big scale?

The address structure lives only while the wireup packet is processed (since it actually points to the packet itself), the connections are not really handled simultaneously but each wireup packet is handled one after another

**Pull request merged**